bcipy/gui/parameters/params_form.py

```diff
@@ -388,6 +400,24 @@ def on_save(self):
             self.changes.update_changes(self.json_file)
             self.changes.repaint()
 
+    def on_save_as(self):
+        """Event handler for saving form data to another parameters file."""
+        options = QFileDialog.Options()
+        options |= QFileDialog.DontUseNativeDialog
+        filename, _ = QFileDialog.getSaveFileName(
+            self,
+            caption='Save As',
+            directory='bcipy/parameters/',
+            filter='JSON files (*.json)',
+            options=options)
```

Comment:
Interestingly, depending on OS this is not guaranteed to be returned with `.json`. I typed in a filename without it and it saved it to a file as `test`, not `test.json`. We should handle the no-extension or bad extension cases.

Reply:
I think the simplest solution is to just append `.json` if the filename does not end with that. That's consistent with how a number of applications handle this (like Microsoft products, I believe).